### PR TITLE
[Dogfood] New semgrep rule to prevent printing on stdout on semgrep codebase

### PR DESCRIPTION
**Pull request by aryx**
This can help https://github.com/returntocorp/semgrep/issues/1630

test plan:
make check

### CODE REVIEW

File: semgrep-core/semgrep.yml [hunk 1]
```diff
@@ -12,3 +12,25 @@ rules:
     message: this will always be true
     languages: [ocaml]
     severity: ERROR
+
+  - id: no-print-in-semgrep
```
Comment (aryx) [inline semgrep-core/semgrep.yml:16]: @mjambon dogfood! Complex OCaml lint rules!

**Comment by aryx** [comment]
ALL GREEN! FINALLY!

**Comment by nbrahms** [comment]
🎉

File: semgrep-core/semgrep.yml (hunk 1 above)
Comment (mjambon) [reply semgrep-core/semgrep.yml:16]: woof

**Pull request merged**